natural_language_processing/text_generation/alpaca/README.md

```diff
@@ -0,0 +1,59 @@
+# Alpaca
+
+This folder contains the script to run Alpaca model on instruction following task in PyTorch framework.
+
+### Dataset
+
+Dataset can be found here: https://github.com/tatsu-lab/stanford_alpaca/blob/main/alpaca_data.json
+
+### Model
+
+To obtain the model, you need to:
+1. Request and download LLaMA v1 model weights https://github.com/facebookresearch/llama/tree/llama_v1
+2. Download the Stanford Alpaca weight diff
+```bash
+git clone https://huggingface.co/tatsu-lab/alpaca-7b-wdiff
+cd alpaca-7b-wdiff
+git lfs install
+git lfs pull
+cd ..
+```
+3. Convert the LLaMA weights to HuggingFace format
+```bash
+git clone https://github.com/huggingface/transformers.git
+python transformers/src/transformers/models/llama/convert_llama_weights_to_hf.py --input_dir <path_to_llama> --model_size 7B --output_dir <output_path>
```

Comment:
```
RuntimeError: Failed to import transformers.models.llama.modeling_llama because of the following error (look up to see its traceback):
No module named 'safetensors'
```

Reply:
```
python3 alpaca-7b-wdiff/transformers/src/transformers/models/llama/convert_llama_weights_to_hf.py --input_dir alpaca_recovered --model_size 7B --output_dir output

FileNotFoundError: [Errno 2] No such file or directory: 'alpaca_recovered/7B/params.json'
```

Reply:
Are you sure you're passing the correct path to this script?  `convert_llama_weights_to_hf.py` is used to convert the original weights from Meta (the ones you have to request access to download them). The `params.json` file should be there.
<img width="217" alt="Screenshot 2023-09-11 at 14 26 10" src="https://github.com/AmpereComputingAI/ampere_model_library/assets/97047305/e2ab2616-f1ff-454d-bf16-ab0fae7f5576">

I can see that you have `alpaca_recovered` in your path, which suggests to me that you used this script on the final Alpaca weights (converted from original weights to HF format and merged with weight diff from Stanford). You should be able to use these weights to run inference with the `run.py` script without any further changes.

tl;dr Are you trying to run step 3 with weights that are the output of step 4?